pipelines/training.py

```diff
@@ -786,44 +793,68 @@ def everything(
-                s3_bucket=s3_bucket,
-                s3_bootleg_prepped_data='None',
-                additional_args=ood_additional_args,
+            image=image,
+            owner=owner,
+            project=project,
+            experiment=experiment,
+            model='%s-ood' % (model,),
+            task_name='ood_task',
+            load_from='None',
+            s3_datadir=train_s3_datadir,
+            dataset_subfolder='ood',
+            genienlp_version=genienlp_version,
+            train_iterations=ood_train_iterations,
+            skip_tensorboard='false',
+            num_gpus='1',
+            valid_set='eval',
+            s3_database_dir='None',
+            train_languages=train_languages,
+            eval_languages=eval_languages,
+            s3_bucket=s3_bucket,
+            s3_bootleg_prepped_data='None',
+            additional_args=ood_additional_args,
         )
 
         pred_op = predicting.prediction_step_small(
-                image=image,
-                owner=owner,
-                genienlp_version=genienlp_version,
-                task_name='ood_task',
-                eval_sets='eval',
-                model_name_or_path=train_op.outputs['s3_model_dir'],
-                s3_input_datadir=train_s3_datadir,
-                s3_database_dir='None',
-                s3_bootleg_prepped_data='None',
-                model_type='None',
-                dataset_subfolder='ood',
-                val_batch_size='4000',
-                additional_args='',
+            image=image,
+            owner=owner,
+            genienlp_version=genienlp_version,
+            task_name='ood_task',
+            eval_sets='eval',
+            model_name_or_path=train_op.outputs['s3_model_dir'],
+            s3_input_datadir=train_s3_datadir,
+            s3_database_dir='None',
+            s3_bootleg_prepped_data='None',
+            model_type='None',
+            dataset_subfolder='ood',
+            val_batch_size='4000',
+            additional_args='',
+        )
+
+    if do_translate:
+        translation_op = translation.dialogue_translation_step(
+            owner=owner,
+            project=project,
+            experiment=experiment,
+            s3_bucket=s3_bucket,
+            s3_datadir=train_s3_datadir,
+            source=source,
+            input_splits=translation_input_splits,
+            model_name_or_path=translation_model_name_or_path,
+            nmt=nmt,
+            src_lang=src_lang,
+            tgt_lang=tgt_lang,
+            image=image,
+            genienlp_version=genienlp_version,
+            genie_version=genie_version,
+            workdir_repo=workdir_repo,
+            workdir_version=workdir_version,
+            thingpedia_developer_key=thingpedia_developer_key,
+            additional_args=translation_additional_args,
         )
 
+        train_s3_datadir = translation_op.outputs['s3_output_datadir']
+
     train_s3_datadir, eval_model = paraphrase_train_fewshot_step(
```

Comment:
I think paraphrasing should also come before translation, because we don't have a paraphraser in the target language. But you experimented with this before, which way does it work best?

(I'm not using paraphrasing right now in production because it's not worth the compute cost, but might reconsider in the future)